File: TP3_Python/Codingan/RAM.py
from Internal import Internal

class RAM(Internal):
    def __init__(self, kapasitas_ram=0, ddr="", nama="", spesifikasi="", merk_internal=""):
        super().__init__(nama, spesifikasi, merk_internal)
        self.kapasitas_ram = kapasitas_ram
        self.ddr = ddr

    def set_kapasitas_ram(self, kapasitas_ram):
        self.kapasitas_ram = kapasitas_ram

    def set_ddr(self, ddr):
        self.ddr = ddr

    def get_kapasitas_ram(self):
        return self.kapasitas_ram

    def get_ddr(self):
        return self.ddr

File: TP3_CPP/Codingan/RAM.cpp
#pragma once
#include "Internal.cpp"
#include <iostream>
#include <string>

using namespace std;

class RAM : public Internal
{
private:
    int KapasitasRAM;
    string DDR;

public:
    RAM()
    {
    }

    RAM(int KapasitasRAM, string DDR, string Nama, string Spesifikasi, string MerkINTERNAL) : Internal(Nama, Spesifikasi, MerkINTERNAL)
    {
        this->KapasitasRAM = KapasitasRAM;
        this->DDR = DDR;
    }

    // Setter
    void setKapasitasRAM(int KapasitasRAM)
    {
        this->KapasitasRAM = KapasitasRAM;
    }

    void setDDR(string DDR)
    {
        this->DDR = DDR;
    }

    // Getter
    int getKapasitasRAM()
    {
        return KapasitasRAM;
    }

    string getDDR()
    {
        return DDR;
    }

    ~RAM()
    {
    }
};
